main.py

```diff
@@ -0,0 +1,229 @@
+import os
+import sys
+import logging
+from transformers import (
+    HfArgumentParser, 
+    Seq2SeqTrainer,
+    Seq2SeqTrainingArguments,
+    AutoConfig,
+    AutoFeatureExtractor,
+    AutoModelForSpeechSeq2Seq,
+    AutoProcessor,
+    AutoTokenizer,
+    set_seed,
+)
+from transformers.trainer_utils import get_last_checkpoint, is_main_process
+from transformers.utils import check_min_version, send_example_telemetry
+from datasets import DatasetDict
+from schemas import ModelArguments, DataTrainingArguments, DataCollatorSpeechSeq2SeqWithPadding
+from src.dataloader import load_datasets_from_config
+from src.metrics import MetricsCalculator, TextNormalizer
+from src.callbacks import ShuffleCallback, EpochProgressCallback
+from loguru import logger
+
+def main():
+    # parse arguments
+    parser = HfArgumentParser((ModelArguments, DataTrainingArguments, Seq2SeqTrainingArguments))
+    
+    if len(sys.argv) == 2 and sys.argv[1].endswith(".json"):
+        model_args, data_args, training_args = parser.parse_json_file(json_file=os.path.abspath(sys.argv[1]))
+    else:
+        model_args, data_args, training_args = parser.parse_args_into_dataclasses()
+        
+    # setting seed for reproducibility
+    set_seed(training_args.seed)
+    
+    # load datasets
+    raw_datasets = DatasetDict()
+    if training_args.do_train:
+        raw_datasets["train"] = load_datasets_from_config(
+            "conf/dataset_config.yaml",
```

Comment:
📝 Bad practice but please let me hardcode the dataset config for now. Will add this to the `args.py` in a later PR. 🙇